### PR TITLE
[15174] Add new CMake option to allow users to enforce use of our third party shared_mutex

### PR DESCRIPTION
**Pull request by MiguelBarro**
## Description

The CMake option **USE_THIRDPARTY_SHARED_MUTEX** is added as **OFF** by default.
By turning it on users can enforce the use of our boost-like implementation of shared_mutex.
Otherwise, the third-party implementation will be used only if the build framework lacks a shared_mutex.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] NA Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- [x] Documentation builds and tests pass locally. 
- [x] New feature has been added to the `versions.md` file (if applicable).
- [x] New feature has been documented/Current behavior is correctly described in the documentation. 

https://github.com/eProsima/Fast-DDS-docs/pull/384

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
Small nitpick

File: versions.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+Version 2.7.1
+-------------
+
+* Added a new CMake option to allow users to enforce use of our third party shared_mutex
```
Comment (jsantiago-eProsima) [inline versions.md:4]:
```suggestion
* Added a new CMake option to allow users to force the use of our third party shared_mutex
```

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Comment by jsantiago-eProsima** [comment]
@richiprosima please test this

**Comment by MiguelBarro** [comment]
@richiprosima Please test this

**Review by jsantiago-eProsima — APPROVED** [review]

**Comment by jsantiago-eProsima** [comment]
LGTM

**Pull request merged**